src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.SkeletonReferenceCache.cs

```diff
@@ -248,15 +252,20 @@ private sealed class SkeletonReferenceSet
             private readonly DeferredDocumentationProvider _documentationProvider;
 
             /// <summary>
-            /// Use WeakReference so we don't keep MetadataReference's alive if they are not being consumed. 
-            /// Note: if the weak-reference is actually <see langword="null"/> (not that it points to null),
-            /// that means we know we were unable to generate a reference for those properties, and future
-            /// calls can early exit.
+            /// The actual assembly metadata produced from the data pointed to in <see cref="_storage"/>.
+            /// </summary>
+            private readonly AsyncLazy<AssemblyMetadata?> _metadata;
```

Comment:
this is a change from before.  previously we might create several instances of AssemblyMetadta if the same compilation was asked for multiple MetadataReferences to it (with different MetadataReferenceProperties).  now, you'll get the same AssemblyMetadata that will be reused in all those cases.

Reply:
This supports the compiler docs which explicitly state:

```
        /// <see cref="Metadata"/> objects may cache information decoded from the PE image.
        /// Reusing <see cref="Metadata"/> instances across metadata references will result in better performance.
```